tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -42,10 +44,18 @@ private DiscoveryService(IApiRequestFactory apiRequestFactory)
 
         public string AgentVersion { get; private set; }
 
-        public static DiscoveryService Create(IApiRequestFactory apiRequestFactory)
+        public static DiscoveryService Create(ImmutableExporterSettings exporterSettings)
         {
             lock (GlobalLock)
             {
+                var apiRequestFactory = AgentTransportStrategy.Get(
+                    exporterSettings,
+                    productName: "discovery",
+                    tcpTimeout: null,
+                    AgentHttpHeaderNames.DefaultHeaders,
+                    () => new DefaultJsonHeaderHelper(),
```

Comment:
You'll need a different one of these too I think

Reply:
Changed here 360b832aff3dc02a2f871a46cb13bf52d25d75fe